### PR TITLE
FindCommand and TestCases

### LINKED ISSUE #58: Implement Find command and basic functionality

### PR DESCRIPTION
**Pull request by shiyao821**
inserted new SampleData for finding
genres now accept '-' hyphens
t/, sn/ and a/ tags are operational, case-insensitive
g/ tags are currently case-sensitive
close #58

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/FindCommandParser.java [hunk 1]
```diff
@@ -20,15 +28,53 @@
      * @throws ParseException if the user input does not conform the expected format
      */
     public FindCommand parse(String args) throws ParseException {
-        String trimmedArgs = args.trim();
+        String trimmedArgs = " 0 " + args.trim();
```
Comment (hoholyin) [inline src/main/java/seedu/address/logic/parser/FindCommandParser.java]: Would you consider declaring a final constant to give meaning to `" 0 "`?

File: src/main/java/seedu/address/logic/parser/FindCommandParser.java [hunk 2]
```diff
@@ -20,15 +28,53 @@
...
+        return new FindCommand(predicate);
+    }
+
+    /**
+     * Parses {@code Collection<String> genres} into a {@code Set<Genre>} if {@code genres} is non-empty.
+     * If {@code genres} contain only one element which is an empty string, it will be parsed into a
+     * {@code Set<Genre>} containing zero genres.
+     */
+    private Optional<Set<Genre>> parseGenresForFind(Collection<String> genres) throws ParseException {
+        assert genres != null;
```
Comment (hoholyin) [inline src/main/java/seedu/address/logic/parser/FindCommandParser.java]: Would you consider using `requireNonNull` here instead?

**Review by hoholyin — COMMENTED** [review]
Just some suggestions to improve readability, but otherwise, new additions seem functional! Good job!

File: src/main/java/seedu/address/logic/parser/FindCommandParser.java (hunk 2 above)
Comment (shiyao821) [reply src/main/java/seedu/address/logic/parser/FindCommandParser.java]: yep! nice catch!

File: src/main/java/seedu/address/logic/parser/FindCommandParser.java (hunk 1 above)
Comment (shiyao821) [reply src/main/java/seedu/address/logic/parser/FindCommandParser.java]: good idea!

File: src/main/java/seedu/address/logic/parser/FindCommandParser.java [hunk 3]
```diff
@@ -1,34 +1,84 @@
...
-import seedu.address.model.book.TitleContainsKeywordPredicate;
+import seedu.address.model.book.BookPredicate;
+import seedu.address.model.genre.Genre;
 
 /**
  * Parses input arguments and creates a new FindCommand object
  */
 public class FindCommandParser implements Parser<FindCommand> {
 
+    private final String preamblePlaceholder = " 0 ";
```
Comment (hoholyin) [inline src/main/java/seedu/address/logic/parser/FindCommandParser.java]: I think we should name constants like `PREAMBLE_PLACEHOLDER` in accordance with the java coding standards for constants 😅

**Review by hoholyin — COMMENTED** [review]
The rest looks good!

File: src/main/java/seedu/address/model/book/BookPredicate.java [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+    }
+
+    public boolean isValid() {
+        return title != null || author != null || serialNumber != null || genres != null;
+    }
+
+    @Override
+    public boolean test(Book book) {
+
+        if (title != null
```
Comment (linyutinglyt) [inline src/main/java/seedu/address/model/book/BookPredicate.java:41]: Good test coverage!

**Pull request merged**